internal/processor/filterhelper/filterhelper.go

```diff
@@ -34,7 +34,48 @@ func NewAttributeValueRaw(value interface{}) (pdata.AttributeValue, error) {
 		return pdata.NewAttributeValueString(val), nil
 	case bool:
 		return pdata.NewAttributeValueBool(val), nil
+	case []interface{}:
+		return fromArray(val), nil
+	case map[string]interface{}:
+		return fromMap(val), nil
 	default:
 		return pdata.AttributeValue{}, fmt.Errorf("error unsupported value type \"%T\"", value)
 	}
 }
+
+func fromArray(val []interface{}) pdata.AttributeValue {
+	arr := pdata.NewAnyValueArray()
+	for _, v := range val {
+		arr.Append(fromVal(v))
+	}
+	av := pdata.NewAttributeValueArray()
+	av.SetArrayVal(arr)
+	return av
+}
+
+func fromMap(v map[string]interface{}) pdata.AttributeValue {
+	m := pdata.NewAttributeMap()
+	for k, v := range v {
+		m.Insert(k, fromVal(v))
+	}
+	m.Sort()
+	av := pdata.NewAttributeValueMap()
+	av.SetMapVal(m)
+	return av
+}
+
+func fromVal(val interface{}) pdata.AttributeValue {
```

Comment:
Can we avoid duplicate code between this func and NewAttributeValueRaw?

Reply:
Noted and updated